pkg/controllers/nodetemplatestatus/controller.go

```diff
@@ -0,0 +1,98 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodetemplatestatus
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/logging"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/patrickmn/go-cache"
+
+	corecontroller "github.com/aws/karpenter-core/pkg/operator/controller"
+	"github.com/aws/karpenter-core/pkg/utils/pretty"
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+)
+
+type Controller struct {
+	kubeClient     k8sClient.Client
+	subnet         *SubnetCollector
+	securityGroups *SecurityGroupsCollector
+}
+
+func NewController(client k8sClient.Client, ec2api ec2iface.EC2API, subnetCache *cache.Cache, SecurityGroupCache *cache.Cache) *Controller {
+	return &Controller{
```

Comment:
Can we try using the `Typed` controller in `karpenter-core/pkg/operator`?

This should allow us to call `Reconcile(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate` with the object already grabbed and typed